Bio/Seq.py

```diff
@@ -2943,6 +2945,45 @@ def join(self, other):
         return Seq(self).join(other)
 
 
+class UndefinedSequenceError(ValueError):
+    """Sequence contents is undefined."""
+
+
+class UndefinedSequenceData:
+    """Stores the length of a sequence with an undefined sequence contents (PRIVATE).
```

Comment:
Do you mean the class to be private? If so leading underscore please?

However, since this seems to be intended to replace the ``UnknownSeq``, ought it not be public?

Reply:
I am not sure. This class really just provides a mechanism telling `Seq` how to behave if the sequence contents is unknown. From that viewpoint, it should be private. That is also safer from a defensive coding viewpoint; if somebody for example subclasses `UndefinedSequenceData`, most likely there is some misunderstanding of how `UndefinedSequenceData` and `Seq` should be used. So I am in favor of making it private.

But if it is private, then users would not be able to create a `Seq` object with an undefined sequence context, at least not from scratch. Probably there is not much demand for that, but perhaps somebody will want to make a qual file from scratch. That would not be possible if `UndefinedSequenceData` is private.

One solution is to allow things like `Seq(6)` or `Seq(None, length=6)` to create an undefined sequence of length 6. Then `Seq` can create an `UndefinedSequenceData` object on the fly, and the `UndefinedSequenceData` class can be private (i.e. a user would never have to deal with an `UndefinedSequenceData` object).